suprb/wrapper.py

```diff
@@ -0,0 +1,55 @@
+import warnings
+
+from suprb import SupRB
+
+"""
+This is a wrapper class for SupRB, where parameters can be set without the need to create objects from deep within suprb.
+
+An instance of SupRB gets instantiated with its default values. Depending on the parameters that are given to SupRBWrapper
+the SupRB instance parameters are adapted. This way one can only set the parameters that they want to change, without the 
+necessity to create the whole object. 
+
+There are two possible ways to set parameters using SupRBWrapper (the below examples result in the same configuration):
+
+1) Have a separate parameter for each nested parameter:
+    - solution_composition__crossover=suprb.optimizer.solution.ga.crossover.NPoint()
+    - solution_composition__crossover__crossover_rate=0.91
+    - solution_composition__crossover__n=3
+
+2) Create an object for a nested parameter and pass that object to the SuprbWrapper:
+    - solution_composition__crossover=suprb.optimizer.solution.ga.crossover.NPoint(crossover_rate=0.91, n=3)
+
+SupRBWrapper has some sanity checks to make sure that the parameter name is in fact part of SupRB. If it is not, it will issue a warning.
+
+There is also the possibility to print the final configuration of the SupRB instance using the parameter print_config
+"""
+class SupRBWrapper():
+    def __new__(self, **kwargs):
+        self.suprb = SupRB()
+        
+        kwargs = dict(sorted(kwargs.items(), key=lambda item: item[0].count("__")))
+        
+        for key, value in kwargs.items():
+            if "print_config" == key:
+                continue
+
+            attribute_string = key.replace("__", ".")
```

Comment:
is this really the standard way to do this?

Reply:
I don't know what the standard way to do this is. The checking and setting of attributes with `hasattr` and `setattr` is definitely the standard way.
Since we have nested parameters, we need to be able to access them somehow. The easiest way that I could come up with, was to iteratively check if the parameter exists, until we arrive at the parameter that we want to change.
So as an example, this is how we would check and change this parameter `layer1.layer2.layer3.param` (we would set the parameter in the wrapper as `layer1__layer2__layer3__param`):

First we check is `layer1` part of `self.suprb` if yes, we continue to `layer2`, if no we print a warning
Then we check is `layer2` part of `layer1` if yes, we continue to `layer3`, if no we print a warning
Then we check is `layer3` part of `layer2` if yes, we continue to `param`, if no we print a warning
Then we check is `param` part of `layer3` if yes, we set the new value to `param,` if no we print a warning

Since we can't have dots (`.`) in a variable name, I needed a different character for the nested parameters. Optuna is using double underscores (`__`) so I thought it makes sense to do the same.